### PR TITLE
feat(Invites): backport INVITE_CREATE and INVITE_DELETE to 11.5-dev

### PR DESCRIPTION
**Pull request by Monbrey**
Backports #3720 

https://github.com/discordjs/discord.js/projects/5#card-31839558

~~I actually tested this one~~

**Status**
- [X] Code changes have been tested against the Discord API, or there are no code changes
- [X] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [X] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/actions/InviteCreate.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const { Events } = require('../../util/Constants');
+
+class InviteCreateAction extends Action {
+  handle(data) {
+    const client = this.client;
+    const inviteData = Object.assign(data, {
+      guild: client.guilds.get(data.guild_id),
+      channel: client.channels.get(data.channel_id),
+    });
+    const invite = new Invite(client, inviteData);
```
Comment (amishshah) [inline src/client/actions/InviteCreate.js]: You should null check here, if the guild or channel you have assigned above is null, then we shouldn't be creating an invite.

File: src/client/actions/InviteDelete.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const { Events } = require('../../util/Constants');
+
+class InviteDeleteAction extends Action {
+  handle(data) {
+    const client = this.client;
+    const inviteData = Object.assign(data, {
+      guild: client.guilds.get(data.guild_id),
+      channel: client.channels.get(data.channel_id),
+    });
+    const invite = new Invite(client, inviteData);
```
Comment (amishshah) [inline src/client/actions/InviteDelete.js]: Same here, null check required.

File: typings/index.d.ts [hunk 3]
```diff
@@ -132,6 +132,7 @@ declare module 'discord.js' {
 		public on(event: 'guildUnavailable', listener: (guild: Guild) => void): this;
 		public on(event: 'guildUpdate', listener: (oldGuild: Guild, newGuild: Guild) => void): this;
 		public on(event: 'guildIntegrationsUpdate', listener: (guild: Guild) => void): this;
+		public on(event: 'inviteCreate' | 'inviteDelete', listener: (invite: Invite) => void): this;
```
Comment (amishshah) [inline typings/index.d.ts]: `inviteDelete` is missing

File: src/client/actions/InviteDelete.js (hunk 2 above)
Comment (Monbrey) [reply src/client/actions/InviteDelete.js]: Resolved in 93be606

File: src/client/actions/InviteCreate.js (hunk 1 above)
Comment (Monbrey) [reply src/client/actions/InviteCreate.js]: Resolved in 93be606

**Pull request merged**